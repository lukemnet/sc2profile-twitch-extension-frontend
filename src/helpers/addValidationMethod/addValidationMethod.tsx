import { YupInstance, Schema } from 'yup';

type SchemaType =
  'mixed'
  | 'string'
  | 'number'
  | 'boolean'
  | 'bool'
  | 'date'
  | 'array'
  | 'object'; 

interface addValidationMethodParams {
  validator: YupInstance;
  schemaType: SchemaType;
  name: string;
  method: (this: Schema<unknown>, message: string, testFn: (testedValues: unknown) => boolean)
    => Schema<unknown>;
}

const addValidationMethod = ({
  validator,
  schemaType,
  name,
  method,
}: addValidationMethodParams) => {
  validator.addMethod(validator[schemaType], name, method);
  return validator;
}

export default addValidationMethod;