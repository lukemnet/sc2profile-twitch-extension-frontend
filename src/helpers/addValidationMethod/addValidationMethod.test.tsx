import * as Yup from 'yup';
import addValidationMethod from './addValidationMethod';

it('adds validation method correctly', () => {
  const yupInstance = Yup;
  addValidationMethod({
    validator: yupInstance,
    schemaType: 'string',
    name: 'testMethod',
    method: function(message) {
      return this.test('testMethod', message, (text) => text.length > 0);
    },
  });

  expect(yupInstance).toMatchSnapshot();
});
