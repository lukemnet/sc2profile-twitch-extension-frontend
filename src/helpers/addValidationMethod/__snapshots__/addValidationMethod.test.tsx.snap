// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`adds validation method correctly 1`] = `
Object {
  "ValidationError": [Function],
  "__esModule": true,
  "addMethod": [Function],
  "array": [Function],
  "bool": [Function],
  "boolean": [Function],
  "date": [Function],
  "isSchema": [Function],
  "lazy": [Function],
  "mixed": [Function],
  "number": [Function],
  "object": [Function],
  "reach": [Function],
  "ref": [Function],
  "setLocale": [Function],
  "string": [Function],
}
`;
